// See README.md for information and build instructions.
//
// Note: START and END tags are used in comments to define sections used in
// tutorials.  They are not part of the syntax for Protocol Buffers.
//
// To get an in-depth walkthrough of this file and the related examples, see:
// https://developers.google.com/protocol-buffers/docs/tutorials

// [START declaration]
syntax = "proto3";
package msg;

//import "google/protobuf/timestamp.proto";
// [END declaration]

// [START java_declaration]
option java_multiple_files = true;
option java_package = "com.example.tutorial.protos";
option java_outer_classname = "AddressBookProtos";
// [END java_declaration]

// [START csharp_declaration]
option csharp_namespace = "Google.Protobuf.Examples.AddressBook";
// [END csharp_declaration]

// [START go_declaration]
option go_package = "github.com/protocolbuffers/protobuf/examples/go/tutorialpb";
// [END go_declaration]

// [START messages]
message Login_info {

  uint32 sign_type = 1;  //sign_in, sign_up, new_game, end_game. not only sign!

  //request
  string name = 2;      //user name

  string passwd = 3;    //password

  uint32 game_seq = 4;  //for sign_type = new_game, end_game

  bool won = 5;         //is server win? for sign_type = end_game

  //response
  uint32 result = 7;    //result status for spercific request, many possible values.....

  string ip = 8;        //only for sign_type = new_game
}
// [END messages]
